index.js

```diff
@@ -33,15 +33,15 @@ function HyperdriveHttp (getArchive) {
     var key = archive
       ? encoding.encode(archive.key)
       : segs.shift()
-    var filename = segs.shift()
+    var filename = segs.join('/')
     var op = 'get'
 
     try {
       // check if we are serving archive at root
       key = key.replace(/\.changes$/, '')
-      encoding.decode(key)
+      encoding.encode(Buffer.from(key, 'hex'))
```

Comment:
`dat-encoding@3.0.1` will now throw on invalid key lenghts. I thought I had a test for this but my test key was `"too short"` which obviously isn't valid hex either, so now I test for both.